### PR TITLE
FABN-1452 NodeSDK add fabric-common doc

### PR DESCRIPTION
**Pull request by harrisob**
Add fabric-common class doc and include updates to tutorials
with the removal of fabric-client details.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -66,12 +44,11 @@ The SDK's list of features include:
   * [customizable persistence store]{@link FabricCAServices}
 
 ### API Reference
-The SDK is made up of 4 top-level modules that can be accessed through the navigation menu **Modules**:
+The SDK is made up of modules that can be accessed through the navigation menu **Modules**:
 * [**fabric-network**]{@link module:fabric-network}: Provides high level APIs for client applications to submit transactions and evaluate queries for a smart contract (chaincode).
 * **api**: Pluggable APIs for application developers to supply alternative implementations of key interfaces used by the SDK. For each interface there are built-in default implementations.
-* **fabric-client**: Provides APIs to interact with the core components of a Hypreledger Fabric-based blockchain network, namely the peers, orderers and event streams.
+* **fabric-common**: Provides APIs to interact with the core components of a Hypreledger Fabric-based blockchain network, namely the peers, orderers and event streams.
```
Comment (heatherlp) [inline docs/index.md]: Hypreledger sounds premature :)

File: docs/tutorials/sign-transaction-offline.md [hunk 2]
```diff
@@ -15,78 +16,62 @@ Javascript `promise` and `async await`.
...
+In most use cases an application will persist the user's credentials including
+the private key and sign transactions for the user. However some business
+scenarios may require higher level of privacy. What if the user wants to
+keep their private key secret and does not trust another system or backend
+server to securely store it and use it?
 
-The `fabric-client` comes with the ability to sign a transaction with an offline private key. By contrast to call `setUserContext()` with the user's identity (which contains the user's private key), an alternative way is to split the `sign a tx` process out of the `fabric-client` and let the application layer choose the place to store the private key, sign the transaction and send the signed transaction back. By this approach, the `fabric-client` does not require the user's private key any more.
+The `fabric-common` comes with the ability to sign a transaction outside of
```
Comment (heatherlp) [inline docs/tutorials/sign-transaction-offline.md]: The `fabric-common` package?

**Review by heatherlp — CHANGES_REQUESTED** [review]
Looks good - just some typos

File: docs/index.md (hunk 1 above)
Comment (harrisob) [reply docs/index.md]: fixed

File: docs/tutorials/handlers.md [hunk 3]
```diff
@@ -15,110 +15,54 @@ and of Node application development, including the use of the
...
-decide to retry, try another end point, or use discovery to complete the
-task.
+decide to retry, try another end point, or use discovery to help
+complete the task.
 
-see {@tutorial discovery} on how the default handlers are
-used with discovery.
+### Service Handler
+fabric-common includes a handler called `DiscoveryHandler`.
+The included handler was designed to be used with discovery to provided automatic
```
[the remark this answers is not in the document]
Comment (harrisob) [reply docs/tutorials/handlers.md]: fixed

File: docs/tutorials/handlers.md [hunk 4]
```diff
@@ -15,110 +15,54 @@ and of Node application development, including the use of the
...
-#### Modified API's that will use handlers
-* `channel.initialize()` - This method has been enhanced to instantiate
-instances of the handlers for use by the channel. The method will get the
-paths from the system configuration settings to create and initialize them.
-* `channel.sendTransactionProposal()` - This method has been enhanced to use
-an `endorsement-handler` if one has been instantiated and initialized.
-* `channel.sendTransaction()` - This method has been enhanced to use
-a `commit-handler` if one has been instantiated and initialized.
+A custom service handler should extend the {@link ServiceHandler} and be passed
+to the `send` method of the service in the optons object attribute `handler`.
```
[the remark this answers is not in the document]
Comment (harrisob) [reply docs/tutorials/handlers.md]: fixed

File: docs/tutorials/sign-transaction-offline.md (hunk 2 above)
Comment (harrisob) [reply docs/tutorials/sign-transaction-offline.md]: fixed

**Review by heatherlp — APPROVED** [review]
LGTM - thanks

**Pull request merged**